modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/controls/NavBar.js

```diff
@@ -36,11 +36,13 @@ function NavBar({creationMenu, showSearch}) {
 					</ManagementToolbar.Item>
 				)}
 
-				{!!sorts.length && Liferay.FeatureFlags['LPD-19465'] && (
-					<ManagementToolbar.Item>
-						<SortDropdown />
-					</ManagementToolbar.Item>
-				)}
+				{!!sorts.length &&
+					sorts.some((sort) => !!sort.label) &&
```

Comment:
This looks like it will work, we're basically distinguishing something that came from DSM by the fact that it has `label` property. The awkward thing about it is that this is "hidden API" from FDS perspective, we're not making it explicit in Java API, [FDSSortItem](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-api/src/main/java/com/liferay/frontend/data/set/model/FDSSortItem.java) and [FDSSortItemBuilder](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-api/src/main/java/com/liferay/frontend/data/set/model/FDSSortItemBuilder.java). That means, if somebody wants to add the `Order` dropdown to FDS using Java API, the only way to do it is to manually build JSON, like we're doing [in fragment](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java#L965-L975). Besides `label`, the same goes for `active` and `default`.

@dsanz @herrardo is the one `Order` menu something that we want to be only related to DSM, or to be a general FDS feature? If it's a general feature, we should show some care to Java API.

Reply:
It makes any sense if we focus on the DSM ordering for now and we extend it to be accesible using Java API later on? Or that would slow us down by a lot?

Reply:
From what I've found in the code these Java APIs end up being used in the `frontend-data-set:headless-display` taglib. And since the `headless-display` taglib renders the same [FrontendDataSet component](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-taglib/src/main/java/com/liferay/frontend/data/set/taglib/servlet/taglib/BaseDisplayTag.java#L277-L279), I'd assume we should add the additional `label` and `default` methods to `FDSSortItem.java` and `FDSSortItemBuilder.java`. (I can update the FDS react component code to not need `active` as a property for the API)

For example `FDSSortItemListBuilder` is used in [ViewChangesDisplayContext.java](https://github.com/liferay/liferay-portal/blob/master/modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/display/context/ViewChangesDisplayContext.java#L292-L298) and then called when using the taglib in [view_changes.jsp](https://github.com/liferay/liferay-portal/blob/master/modules/apps/change-tracking/change-tracking-web/src/main/resources/META-INF/resources/publications/view_changes.jsp#L42).

Reply:
> I'd assume we should add the additional label and default methods to FDSSortItem.java and FDSSortItemBuilder.java. (I can update the FDS react component code to not need active as a property for the API)

Sounds right, few additional getters and setters.

Reply:
After a conversation with @thektan the other day, we decided this can be done in a separate task. Reason is that the FDSSortItem is actually extending `HashMap<String, Object>` which means that in practise, java usages could invoke `fdsSortItem.put("label", "value");` and that would be serialized.

This works under the assumption that existing usages of Java API to create `FDSSortItems` are not affected at all by the behavior we're adding here

This way we can defer to a future tech task the addition of the needed machinery in terms of new Java methods, builder enhancements to account for them, etc. That would make Java dev's life easier. I doubt we need to do this in the epic, but we can try depending on available bandwidth